src/main/java/com/minecolonies/util/ExperienceUtils.java

```diff
@@ -21,25 +23,31 @@ private ExperienceUtils()
     }
 
     /**
-     * Calculates the xp needed for the next level.
-     * @param currentLevel the currentLevel of the citizen
-     * @return the xp in int
-     */
-    public static int getXPNeededForNextLevel(int currentLevel)
-    {
-        return EXPERIENCE_MULTIPLIER * (currentLevel+1) * (currentLevel+1);
-    }
-
-    /**
      * Calculates how much percent of the current level have been completed
+     *
      * @param experience the current amount of xp
-     * @param level the current level
+     * @param level      the current level
      * @return the percentage
      */
     public static double getPercentOfLevelCompleted(double experience, int level)
     {
-        return PERCENT_MULTIPLIER-((getXPNeededForNextLevel(level) - experience) / getXPNeededForNextLevel(level)) * PERCENT_MULTIPLIER;
+        return PERCENT_MULTIPLIER
+                 - ((getXPNeededForNextLevel(level)
+                       - experience)
+                      / getXPNeededForNextLevel(level))
+                     * PERCENT_MULTIPLIER;
     }
 
-
+    /**
+     * Calculates the xp needed for the next level.
+     *
+     * @param currentLevel the currentLevel of the citizen
+     * @return the xp in int
+     */
+    public static double getXPNeededForNextLevel(int currentLevel)
+    {
+        return EXPERIENCE_MULTIPLIER
+                 * (currentLevel + 1)
+                 * (currentLevel + 1);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Cast one of the operands of this multiplication operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2184)